testkit/src/checkpoint_db.rs

```diff
@@ -73,31 +74,50 @@ pub struct CheckpointDbHandler<T> {
 }
 
 impl<T: Database> CheckpointDbHandler<T> {
+    /// Set a checkpoint for future rollback
+    ///
+    /// # Panics
+    ///
+    /// - Panics if another checkpoint was set before and have not been rolled back to.
+    pub fn checkpoint(&self) {
+        self.inner
+            .write()
+            .expect("Cannot lock CheckpointDb for checkpoint")
+            .checkpoint();
+    }
+
     /// Rolls back this database by undoing the latest `count` `merge()` operations.
     ///
     /// # Panics
     ///
-    /// - Panics if more operations are attempted to be reverted than the number of operations
-    ///   in the DB journal.
-    pub fn rollback(&self, count: usize) {
+    /// - Panics if there is no checkpoint to rollback to.
+    pub fn rollback(&self) {
         self.inner
             .write()
             .expect("Cannot lock CheckpointDb for rollback")
-            .rollback(count);
+            .rollback();
     }
 }
 
 #[derive(Debug)]
+// Journal is used here to track patches that need to be applied (in order
+// that they appear) to db for it to become what it was on checkpoint.
+//
+// Insertion should happen to the front of the journal,
+// application of patches in the journal should happen
+// sequentially in order that they are in the VecDeque.
```

Comment:
`Insertion should occur at the front of the journal, while patches should apply sequentially in the order in which they are in the VecDeque.`

I'm not native, but I think it'll be better. Contact me if our opinions mismatch.